### ISSUE TITLE
error --unsupported Microsoft Visual Studio version! Only the versions between 2013 and 2017 are supported 

### ISSUE THREAD

**Issue by abdou31** [opencv/opencv#17594]
- OpenCV => 4.4.0
- Operating System / Platform => Windows 10 64 Bit
- Compiler => Visual Studio 2019
- Cuda: 10.0
- Cudnn:7.5
- Gpu: Nvidia Geforce 840 m

##### Detailed description

Hello, I'm trying to install OpenCV from sources using VS 2019 after generating a build folder using CMake-GUI, I'm getting two errors, 

The first is :
`Erreur	MSB6006  "cmd.exe" exited with code -1073741819.	ZERO_CHECK`

The second error:

`C1189	#error:  -- unsupported Microsoft Visual Studio version! Only the versions between 2013 and 2017 (inclusive) are supported!	opencv_core	C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\include\crt\host_config.h	143`

![image](https://user-images.githubusercontent.com/19480228/85186967-59d51300-b29c-11ea-92bf-1eb57d3a2952.png)

how can I fix those errors???

**Comment by YashasSamaga** [comment]
You have to install the previous version of the toolset (v141). Open Visual Studio setup and look for 2017 v141 toolset.

Open Configuration > General and set "Platform Toolset" to "Visual Studio 2017 v141" (or something similar).

**Comment by abdou31** [comment]
I have already installed both of toolset ( v141 and v140 ) and I’v tested but I’m still getting this error.
I’m trying a solution deleting the second part of the statement ( || _msvc >= 1920 ) in host_config.h file.
It seems work but I’m getting a lot of warnings messages( 2000 )

Do you can check if this can be the right solution or not ?

**Comment by YashasSamaga** [comment]
I can't find any documentation for `_msvc`. I will assume that it behaves like `_MSC_VER`. MSVC version should be 1916 or below if you are using the v140 platform toolset (and that check in `host_config.h` should fail). Can you check the configuration again if you have selected the correct toolset?

[VS 2019 support was added in CUDA 10.1 Update 2](https://docs.nvidia.com/cuda/archive/10.1/cuda-toolkit-release-notes/index.html). You can also upgrade to 10.1 Update 2 and higher.

**Comment by abdou31** [comment]
Sorry, you are right, I wrote it in the incorrect format.
I'm using VS 2019, CUDA 10 and I'm setting Visual Studio toolset 2017 as platform tools.
The change that I've made works fine for me. I don't have any idea if I will encounter an error in the future because I have got many warnings messages.

The tutorial seems to work with OpenCV and My GPU with <= 10 fps. but it works in the end and this means that the OpenCV is installed correctly.

Thanks for your help.
